dwave/optimization/include/dwave-optimization/nodes/numbers.hpp

```diff
@@ -46,14 +46,38 @@ class NumberNode : public ArrayOutputMixin<ArrayNode>, public DecisionNode {
 
     // Overloads required by the Node ABC *************************************
 
+    // Initialize the state. Defaults to 0 if 0 is in range, otherwise defaults
+    // to lower bound.
     void initialize_state(State& state) const override;
-    void initialize_state(State& state, RngAdaptor& rng) const;
-
-    template <std::ranges::range R>
-    void initialize_state(State& state, R&& number_data) const;
 
+    // Initialize a state from an existing container, taking ownership.
     void initialize_state(State& state, std::vector<double>&& number_data) const;
 
+    // Initialize a state from an existing container, making a copy.
+    template<std::ranges::range R>
+    void initialize_state(State& state, const R& values) const {
+        return initialize_state(state, std::vector<double>(values.begin(), values.end()));
+    }
+
+    // Initialize the state of the node randomly
+    template <std::uniform_random_bit_generator Generator>
```

Comment:
Why keep this version of the method around?

Reply:
We use it in testing. So I was trying to limit changes.